File: HumanEvalJava/src/main/java/original/id_18.java
package original;


class HowManyTimes {
    /**
     * Find how many times a given substring can be found in the original string. Count overlapping cases.
     *
     * > howManyTimes("", "a")
     * 0
     * > howManyTimes("aaa", "a")
     * 3
     * > howManyTimes("aaaa", "aa")
     * 3
     * > howManyTimes("", "x")
     * 0
     * > howManyTimes("xyxyxyx", "x")
     * 4
     * > howManyTimes("cacacacac", "cac")
     * 4
     * > howManyTimes("john doe", "john")
     * 1
     */
    public static int howManyTimes(String string, String substring) {
        int count = 0;
        int index = 0;
        while (index < string.length()) {
            index = string.indexOf(substring, index);
            if (index == -1) {
                break;
            }
            count++;
            index++;
        }
        return count;
    }


}

File: HumanEvalPython/scenario2/id_18.py


def how_many_times(string: str, substring: str) -> int:
    """ Find how many times a given substring can be found in the original string. Count overlaping cases.
    
    
    """

    times = 0

    for i in range(len(string) - len(substring) + 1):
        if string[i:i+len(substring)] == substring:
            times += 1

    return times
